### PR TITLE
Feature:Home-Display all events from authenticated user

### PR DESCRIPTION
**Pull request by SilviaTrujillano23**
## Describe your changes
 
- Render events from Home
- Display "subscribed" or "register" according to event subscription state from authenticated user
- Validate event-user from endpoint 

## Issue ticket number and link
https://trello.com/c/c5LavIZ3

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update: 
- [ ] Create storybook for components involved

Home when user is authen
Screenshot 2023-07-06 at 18 30 58
ticated

### CODE REVIEW

File: src/shared/api/endpoints/authentication.ts [hunk 1]
```diff
@@ -21,19 +21,23 @@ function Authentication() {
...
-        if (user?.uid) {
-          resolve({ isAuth: true, userUid: user.uid })
-        } else {
-          reject({ isAuth: false, userUid: '' })
-        }
-      })
-    })
+
+    return new Promise<{ isAuth: boolean; userUid: string; email: string }>(
+      function (resolve, reject) {
```
Comment (jose-sal-y-rosas) [inline src/shared/api/endpoints/authentication.ts:26]: Looks fine as anonymous function but I think you can use arrow function here or it is IIFE?

File: src/shared/api/endpoints/conferences.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ import config from '../../../environment/environment'
 
 function ConferenceAPI() {
   const getAll = async (): Promise<Conference[]> => {
+    console.log('conference api', config.eventCollectionName)
```
Comment (jose-sal-y-rosas) [inline src/shared/api/endpoints/conferences.ts:7]: eppa! console.log here ¬¬

**Review by jose-sal-y-rosas — CHANGES_REQUESTED** [review]
Please remove console.log

File: src/shared/api/endpoints/authentication.ts (hunk 1 above)
Comment (SilviaTrujillano23) [reply src/shared/api/endpoints/authentication.ts:26]: Now modified to arrow function.Thanks

File: src/shared/api/endpoints/conferences.ts (hunk 2 above)
Comment (SilviaTrujillano23) [reply src/shared/api/endpoints/conferences.ts:7]: Now removed

**Review by FelipeLosada1996 — APPROVED** [review]

File: src/shared/api/endpoints/authentication.ts [hunk 3]
```diff
@@ -21,19 +21,23 @@ function Authentication() {
...
+
+    return new Promise<{ isAuth: boolean; userUid: string; email: string }>(
+      (resolve, reject) => {
+        const auth = getAuth(getFirebaseApp())
+        auth.onAuthStateChanged((user) => {
+          if (user?.uid) {
+            resolve({
+              isAuth: true,
+              userUid: user.uid,
+              email: user.email as string,
```
Comment (jlora-mck) [inline src/shared/api/endpoints/authentication.ts:33]: why we need to cast as string ? also we can destructure user

Comment (SilviaTrujillano23) [reply src/shared/api/endpoints/authentication.ts]: Casting as string otherwise email return null, user is now destructured.

**Review by jlora-mck — APPROVED** [review]

**Pull request merged**